enumerable_methods.rb

```diff
@@ -0,0 +1,125 @@
+# frozen_string_literal: true
+
+module Enumerable
+  def my_each
+    return to_enum unless block_given?
+
+    index = 0
+    while index < size
+      yield(self[index])
+      index += 1
+    end
+    self
+  end
+
+  def my_each_with_index
+    return to_enum unless block_given?
+
+    index = 0
+    while index < size
+      yield(self[index], index)
+      index += 1
+    end
+    self
+  end
+
+  def my_select
+    return to_enum unless block_given?
+
+    selected = []
+    my_each do |element|
+      next unless yield(element)
+
+      selected << element
+    end
+    selected
+  end
+
+  def my_all?(arg = nil)
+    if block_given?
+      my_each { |element| return false unless yield(element) }
+    elsif arg.nil
```

Comment:
Test your method against the following cases:

```
%w[ant bear cat].all?(/t/)                        #=> false
[1, 2i, 3.14].all?(Numeric)                       #=> true
[nil, true, 99].all?                              #=> false
[].all?                                           #=> true
```

- [ ] my_all? when no block or argument is given returns true when none of the collection members are false or nil

Your method is failing, but the expected result would be:

```
irb(main):120:0> [1,2,3].all?
=> true
```
- [ ]  my_all? when a class is passed as an argument returns true if all of the collection is a member of such class

- [ ] my_all? when a Regex is passed as an argument returns true if all of the collection matches the Regex

- [ ] my_all? when a pattern other than Regex or a Class is given returns true if all of the collection matches the pattern

``` 
irb(main):127:0> ["abc",1].all?(1)
=> false
irb(main):128:0> [1,1].all?(1)
=> true
irb(main):129:0> ["abc","abc"].all?(1)
=> false
irb(main):130:0> ["abc","abc"].all?("abc")
=> true
```